### PR TITLE
Animates speech bubble

### PR DESCRIPTION
**Pull request by MoXuT0**
## About The Pull Request

Stolen from old hippiestation. ~Only works when you send speech for now, need to figure out a way to make emotes use this too without fucking up typing indicator.~ Will do another PR on emote overlays when I figure out how to do it correctly.

## Changelog
:cl:
add: Speech bubble is now animated.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -199,7 +199,7 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 			speech_bubble_recipients.Add(M.client)
 	var/image/I = image('icons/mob/talk.dmi', src, "[bubble_type][say_test(message)]", FLY_LAYER)
 	I.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA
-	INVOKE_ASYNC(GLOBAL_PROC, /.proc/flick_overlay, I, speech_bubble_recipients, 30)
+	animate_speechbubble(I, speech_bubble_recipients, TYPING_INDICATOR_LIFETIME)
```
Comment (psykzz) [inline code/modules/mob/living/say.dm]: Should this not still be invoked async?

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -1179,4 +1179,18 @@ var/list/WALLITEMS = list(
...
+
+/proc/animate_speechbubble(image/I, list/show_to, duration)
+    var/matrix/M = matrix()
+    M.Scale(0,0)
+    I.transform = M
+    I.alpha = 0
+    for(var/client/C in show_to)
+        C.images += I
+    animate(I, transform = 0, alpha = 255, time = 5, easing = ELASTIC_EASING)
+    sleep(duration - 5)
```
Comment (DominikPanic) [inline code/__HELPERS/unsorted.dm]: Use a callback

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -1179,4 +1179,18 @@ var/list/WALLITEMS = list(
...
+    animate(I, alpha = 0, time = 5, easing = EASE_IN)
+    sleep(5)
```
Comment (DominikPanic) [inline code/__HELPERS/unsorted.dm]: Same

File: code/modules/mob/living/say.dm (hunk 1 above)
Comment (DominikPanic) [reply code/modules/mob/living/say.dm]: Ye it should

File: code/__HELPERS/unsorted.dm [hunk 4]
```diff
@@ -1179,4 +1179,18 @@ var/list/WALLITEMS = list(
...
+
+/proc/animate_speechbubble(image/I, list/show_to, duration)
+    var/matrix/M = matrix()
+    M.Scale(0,0)
+    I.transform = M
+    I.alpha = 0
+    for(var/client/C in show_to)
+        C.images += I
+    animate(I, transform = 0, alpha = 255, time = 5, easing = ELASTIC_EASING)
+    addtimer(CALLBACK(GLOBAL_PROC, /.proc/fade_out, I), duration - 5)
```
Comment (Rohesie) [inline code/__HELPERS/unsorted.dm]:
```suggestion
    addtimer(CALLBACK(GLOBAL_PROC, /.proc/fade_out, I), duration - 0.5 SECONDS)
```
Same thing, just a little more clarity.

**Pull request merged**